ui/src/data_explorer/actions/view/index.js

```diff
@@ -37,14 +37,31 @@ export const groupByTime = (queryId, time) => ({
   },
 })
 
-// all fields implicitly have a function applied to them, so consequently
-// we need to set the auto group by time
-export const toggleFieldWithGroupByInterval = (
-  queryID,
-  fieldFunc
-) => dispatch => {
+export const fill = (queryId, value) => ({
+  type: 'DE_FILL',
+  payload: {
+    queryId,
+    value,
+  },
+})
+
+// all fields implicitly have a function applied to them by default, unless
+// it was explicitly removed previously, so set the auto group by time except
+// under that removal condition
+export const toggleFieldWithGroupByInterval = (queryID, fieldFunc) => (
+  dispatch,
+  getState
+) => {
   dispatch(toggleField(queryID, fieldFunc))
-  dispatch(groupByTime(queryID, DEFAULT_DATA_EXPLORER_GROUP_BY_INTERVAL))
+  // toggleField determines whether to add a func, so now check state for funcs
+  // presence, and if present then apply default group by time
+  const newFieldFunc = getState().dataExplorerQueryConfigs[queryID].fields.find(
+    ({field}) => field === fieldFunc.field
+  )
+  // newFieldFunc could be undefined if it was toggled for removal
+  if (newFieldFunc && newFieldFunc.funcs.length) {
```

Comment:
Ok, I understand. I added it because it documents an unintuitive reason as to why `updatedFieldFunc` may be undefined; the logic only communicates _that_ it may `undefined`, but not why. I'm gonna leave it to err on the side of clarity.